javaslang/src/main/java/javaslang/collection/Vector.java

```diff
@@ -611,15 +655,10 @@ private Vector(int indexShift, HashArrayMappedTrie<Integer, T> trie) {
     public Vector<T> drop(int n) {
         if (n <= 0) {
             return this;
-        }
-        if (n >= length()) {
+        } else if (n >= length()) {
             return empty();
         }
-        HashArrayMappedTrie<Integer, T> trie = HashArrayMappedTrie.empty();
-        for (int i = n; i < length(); i++) {
-            trie = trie.put(i - n, get(i));
-        }
-        return wrap(trie);
+        throw new UnsupportedOperationException();
```

Comment:
I think we should leave this PR open until Vector is finished. The master branch should contain production ready code.